frontend/src/main/scala/bloop/bsp/BspServer.scala

```diff
@@ -59,6 +61,9 @@ object BspServer {
     }
   }
 
+  private final val connectedBspClients =
+    new ConcurrentHashMap[AbsolutePath, ClientInfo.BspClientInfo]()
```

Comment:
Can this index represent multiple BSP clients in the same directory?

Reply:
Nope, good catch, I forgot to take care of this case